tools/CheckFiles.sh

```diff
@@ -107,38 +112,69 @@ find ./ -type f                                                              \
      ! -name "*.patch"                                                       \
      ! -name "*LICENSE.*"                                                    \
      ! -name "*.clang-format"                                                \
+     ! -path "./build*"                                                      \
     | xargs grep -L "^.*Distributed under the MIT License"`
 if [[ $files_without_license != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     echo "Did not find a license in these files:"
     echo "$files_without_license"
-    exit 1
+    found_error=1
 fi
 
 ###############################################################################
 # Check for tests using Catch's TEST_CASE instead of SPECTRE_TEST_CASE
 found_test_case=`
-find ./ -type f -name '*.[ch]pp' \
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
     | xargs grep --with-filename -n "^TEST_CASE"`
 if [[ $found_test_case != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     echo "Found occurrences of TEST_CASE, must use SPECTRE_TEST_CASE:"
     echo "$found_test_case"
-    exit 1
+    found_error=1
 fi
 
 ###############################################################################
 # Check for tests using Catch's Approx, which has a very loose tolerance
 found_bad_approx=`
-find ./ -type f -name '*.[ch]pp' \
+find ./ -type f -name '*.[ch]pp' -not -path './build*' \
     | xargs grep --with-filename -n "Approx("`
 if [[ $found_bad_approx != "" ]]; then
     echo "This script can be run locally from any source dir using:"
     echo "SPECTRE_ROOT/tools/CheckFiles.sh"
     printf "Found occurrences of Approx, must use approx from " \
            "SPECTRE_ROOT/tests/Unit/TestHelpers.hpp instead:\n"
     echo "$found_bad_approx"
+    found_error=1
+fi
+
+###############################################################################
+# Check for Doxygen group inclusions on the starting line
+found_incorrect_ingroup_syntax=`
```

Comment:
rename variable to found_bad_doxygen_syntax and update comment since this will error on anything on the starting line